src/App.tsx

```diff
@@ -1,24 +1,98 @@
 import React from 'react';
 import './App.scss';
+import './GoodsList.scss';
 
 const goodsFromServer: string[] = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  'dumplings',
+  'carrot',
+  'eggs',
+  'ice cream',
+  'apple',
+  'bread',
+  'fish',
+  'honey',
+  'jam',
+  'garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[];
+}
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['jam'],
+  };
+
+  addHandler = (good: string) => {
+    this.setState(({ selectedGoods }) => ({
+      selectedGoods: [...selectedGoods, good],
+    }));
+  };
+
+  removeHandler = (good: string) => {
+    this.setState(({ selectedGoods }) => ({
+      selectedGoods: selectedGoods.filter((g) => g !== good),
+    }));
+  };
+
+  clearHandler = () => {
+    this.setState(() => ({
+      selectedGoods: [],
+    }));
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    const formattedList = selectedGoods.length === 1
+      ? selectedGoods[0]
+      : (`${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods.slice(-1)}`);
+
+    const includes = (good: string) => selectedGoods.includes(good);
+    const text = (good: string) => (includes(good) ? 'remove' : 'add');
+    const method = (good: string) => (includes(good)
+      ? this.removeHandler(good)
+      : this.addHandler(good)
+    );
+
+    return (
+      <div className="App">
+        <h1 className="App__title">
+          {!selectedGoods.length
+            ? 'no goods selected'
+            : (`selected: ${formattedList}`)}
+        </h1>
+        <ul className="GoodsList">
+          {goodsFromServer.map((good) => (
+            <li
+              key={good}
+              className={`GoodsList__item ${includes(good) && 'GoodsList__item--selected'}`}
+            >
+              <span>{good}</span>
+              <button
+                type="button"
+                className="GoodsList__item-button"
+                onClick={() => {
+                  method(good);
+                }}
+              >
+                {text(good)}
+              </button>
+            </li>
+          ))}
+        </ul>
+        {!!selectedGoods.length && (
```

Comment:
Краще явно приводити до типу
```suggestion
        {Boolean(selectedGoods.length) && (
```